sqld/src/admin_api.rs

```diff
@@ -101,7 +103,15 @@ async fn handle_create_namespace_with_dump<F: MakeNamespace>(
 async fn handle_create_namespace<F: MakeNamespace>(
     State(app_state): State<Arc<AppState<F>>>,
     Path(namespace): Path<String>,
-) -> Result<(), crate::error::Error> {
+) -> crate::Result<()> {
     app_state.namespaces.create(namespace.into(), None).await?;
     Ok(())
 }
+
+async fn handle_delete_namespace<F: MakeNamespace>(
+    State(app_state): State<Arc<AppState<F>>>,
+    Path(namespace): Path<String>,
+) -> crate::Result<()> {
+    app_state.namespaces.destroy(namespace.into()).await?;
```

Comment:
I've noticed that while we convert namespace to `Bytes` here, deeper in the call stack it's sometimes used as `String` again ie. when performing potentially non-safe UTF-8 conversion in order to include namespace in error message.

Maybe it would be better for these methods to take `String` as a parameter and only convert it to `Bytes` when necessary?

Reply:
Initially I wanted it to be an Arc<str> because we keep cloning it everywhere, but this is not supported by tonic, so I had to resolve to use bytes instead